ts/src/vector/filter/sequenceSelectionVector.spec.ts

```diff
@@ -0,0 +1,202 @@
+            expect(customVector.getIndex(0)).toBe(10);
+            expect(customVector.getIndex(1)).toBe(15);
+
+            // After setting, array should be materialized
+            customVector.setIndex(2, 999);
+            expect(customVector.getIndex(2)).toBe(999);
+
+            // Other values should still be correct
+            expect(customVector.getIndex(0)).toBe(10);
+            expect(customVector.getIndex(1)).toBe(15);
+        });
+
+        it("Should return Index out of bounds", () => {
+            expect(() => seqVector.setIndex(-1, 0)).toThrowError("Index out of bounds");
+            expect(() => seqVector.setIndex(25, 52)).toThrowError("Index out of bounds");
+        });
+
+        it("Should preserve materialized values after multiple sets", () => {
+            seqVector.setIndex(0, 100);
+            seqVector.setIndex(1, 200);
+            seqVector.setIndex(2, 300);
+
+            expect(seqVector.getIndex(0)).toBe(100);
+            expect(seqVector.getIndex(1)).toBe(200);
+            expect(seqVector.getIndex(2)).toBe(300);
+        });
+    });
+
+    describe("setLimit Test", () => {
+        it("Should set limit", () => {
+            seqVector.setLimit(250);
+            expect(seqVector.limit).toBe(250);
+            seqVector.setLimit(0);
+            expect(seqVector.limit).toBe(0);
+            seqVector.setLimit(-125);
+            expect(seqVector.limit).toBe(-125);
+        });
+
+        it("Should change limit independently of capacity", () => {
+            expect(seqVector.capacity).toBe(5);
+            seqVector.setLimit(3);
+            expect(seqVector.limit).toBe(3);
+            expect(seqVector.capacity).toBe(5);
+        });
+    });
+
+    describe("selectionValues Test", () => {
+        it("Should return selectionVector", () => {
+            const values = seqVector.selectionValues();
+            expect(values).toStrictEqual([0, 1, 2, 3, 4]);
+        });
+
+        it("Should return empty array for empty vector", () => {
+            const emptyVector = new SequenceSelectionVector(0, 1, 0);
+            expect(emptyVector.selectionValues()).toStrictEqual([]);
+        });
+
+        it("Should return correct values with custom base and delta", () => {
+            const customVector = new SequenceSelectionVector(10, 5, 5);
+            expect(customVector.selectionValues()).toStrictEqual([10, 15, 20, 25, 30]);
+        });
+
+        it("Should return correct values with negative delta", () => {
+            const negVector = new SequenceSelectionVector(100, -10, 5);
+            expect(negVector.selectionValues()).toStrictEqual([100, 90, 80, 70, 60]);
+        });
+
+        it("Should cache materialized array", () => {
+            const values1 = seqVector.selectionValues();
+            const values2 = seqVector.selectionValues();
+            expect(values1).toBe(values2); // Same reference
+        });
+
+        it("Should return materialized array after setIndex", () => {
```

Comment:
I think there's no need to mention the `materialized` part of it, the test simply makes sure that after setIndex the selectionValues still works as expected.